modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java

```diff
@@ -0,0 +1,318 @@
+import java.util.Map;
+import java.util.ResourceBundle;
+
+import javax.servlet.jsp.JspException;
+import javax.servlet.jsp.JspWriter;
+
+/**
+ * @author Julien Castelain
+ */
+public class PaginationBarTag extends BaseContainerTag {
+
+	@Override
+	public int doStartTag() throws JspException {
+		setAttributeNamespace(_ATTRIBUTE_NAMESPACE);
+
+		PaginationBarLabels labels = getLabels();
+
+		if (labels == null) {
+			setLabels(
+				new PaginationBarLabels(
+					"Showing {0} to {1} of {2}", "{0} items", "{0} items"));
+		}
+
+		return super.doStartTag();
+	}
+
+	public Integer getActiveDelta() {
+		return _activeDelta;
+	}
+
+	public Integer getActivePage() {
+		return _activePage;
+	}
+
+	public List<PaginationBarDelta> getDeltas() {
+		return _deltas;
+	}
+
+	public List<Integer> getDisabledPages() {
+		return _disabledPages;
+	}
+
+	public Integer getEllipsisBuffer() {
+		return _ellipsisBuffer;
+	}
+
+	public PaginationBarLabels getLabels() {
+		return _labels;
+	}
+
+	public Integer getTotalItems() {
+		return _totalItems;
+	}
+
+	public void setActiveDelta(Integer activeDelta) {
+		_activeDelta = activeDelta;
+	}
+
+	public void setActivePage(Integer activePage) {
+		_activePage = activePage;
+	}
+
+	public void setDeltas(List<PaginationBarDelta> deltas) {
+		_deltas = deltas;
+	}
+
+	public void setDisabledPages(List<Integer> disabledPages) {
+		System.out.println("Disabled pages set");
+		_disabledPages = disabledPages;
+	}
+
+	public void setEllipsisBuffer(Integer ellipsisBuffer) {
+		_ellipsisBuffer = ellipsisBuffer;
+	}
+
+	public void setLabels(PaginationBarLabels labels) {
+		_labels = labels;
+	}
+
+	public void setTotalItems(Integer totalItems) {
+		_totalItems = totalItems;
+	}
+
+	@Override
+	protected void cleanUp() {
+		super.cleanUp();
+
+		_activeDelta = null;
+		_activePage = null;
+		_deltas = null;
+		_disabledPages = null;
+		_ellipsisBuffer = null;
+		_labels = null;
+		_totalItems = null;
+	}
+
+	@Override
+	protected String getHydratedModuleName() {
+		return "frontend-taglib-clay/PaginationBar";
+	}
+
+	@Override
+	protected Map<String, Object> prepareProps(Map<String, Object> props) {
+		props.put("activeDelta", _activeDelta);
+		props.put("activePage", _activePage);
+		props.put("deltas", _deltas);
+		props.put("disabledPages", _disabledPages);
+		props.put("ellipsisBuffer", _ellipsisBuffer);
+		props.put("labels", _labels);
+		props.put("totalItems", _totalItems);
+
+		return super.prepareProps(props);
+	}
+
+	@Override
+	protected int processStartTag() throws Exception {
+		super.processStartTag();
+
+		JspWriter jspWriter = pageContext.getOut();
+
+		jspWriter.write("<div class=\"pagination-bar\"><div class=\"");
+		jspWriter.write("dropdown pagination-items-per-page\">");
+
+		ResourceBundle resourceBundle = TagResourceBundleUtil.getResourceBundle(
+			pageContext);
+
+		Integer activeDelta = getActiveDelta();
+
+		if (activeDelta == null) {
+			PaginationBarDelta defaultDelta = getDeltas().get(0);
```

Comment:
usually we use the type for variable names

Reply:
Do you mean that instead of 

```java
PaginationBarDelta defaultDelta = ...
```

It should be 

```java
PaginationBarDelta paginationBarDeltaDefaultDelta;
```

?

Reply:
Just paginationBarDelta. It you want to add the default it should finish with the type: defaultPaginationBarDelta

Reply:
But we can try to leave it this way (defaultDelta) to see if he checks the taglib code in the same way as the other modules :)

Reply:
Ok thanks,  I'll stick to the conventions to avoid any problems :+1:

Reply:
I updated that in [`a0f87e1`](https://github.com/liferay-frontend/liferay-portal/pull/1224/commits/a0f87e1041b6acc4986ffc920774f5f38e34b77b) along with some other changes :see_no_evil: